bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/votable/VoTableParser.scala

```diff
@@ -49,29 +49,34 @@ object VoTableParser extends VoTableParser {
   /**
    * parse takes an input stream and attempts to read the xml content and convert it to a VoTable resource
    */
-  def parse(catalog: CatalogName, is: InputStream): CatalogResult = {
-    // Load in memory (Could be a problem for large responses)
-    val xmlText = Source.fromInputStream(is, "UTF-8").getLines().mkString
-
-    catalog match {
-      case SIMBAD =>
-        // Simbad is a special case as it is not fully votable-compliant.
-        // We want to catch some errors at this level to simplify the parse method
-        // that assumes we are votable compliant
-        val xml = XML.loadString(xmlText)
-        if (SimbadAdapter.containsExceptions(xml)) {
-          \/.left(ValidationError(catalog))
-        } else {
-          \/.right(parse(xml))
-        }
-      case _ if validate(xmlText).isLeft => \/.left(ValidationError(catalog))
-      case _                             => \/.right(parse(XML.loadString(xmlText)))
+  def parse(catalog: CatalogName, is: InputStream): CatalogResult =
+    (CatalogAdapter.forCatalog(catalog) \/> UnknownCatalog).flatMap { adapter =>
+      // Load in memory (Could be a problem for large responses)
+      val xmlText = Source.fromInputStream(is, "UTF-8").getLines().mkString
+
+      catalog match {
+        case CatalogName.SIMBAD =>
+          // Simbad is a special case as it is not fully votable-compliant.
+          // We want to catch some errors at this level to simplify the parse method
+          // that assumes we are votable compliant
+          val xml = XML.loadString(xmlText)
+          if (CatalogAdapter.Simbad.containsExceptions(xml)) {
+            \/.left(ValidationError(catalog))
+          } else {
+            \/.right(parse(adapter, xml))
+          }
+        case _ if validate(xmlText).isLeft => \/.left(ValidationError(catalog))
+        case _                             => \/.right(parse(adapter, XML.loadString(xmlText)))
+      }
     }
-  }
 }
 
 // A CatalogAdapter improves parsing handling catalog-specific options like parsing magnitudes and selecting key fields
 sealed trait CatalogAdapter {
+
+  /** Identifies the catalog to which the adapter applies. */
+  def catalog: CatalogName
```

Comment:
The adapters now know which catalog they correspond to so that we can look them up easily.